### ISSUE TITLE
CloudConfigurationManager ALLOWS application to fall back to local appSettings

### ISSUE THREAD

**Issue by MichaelCommo** [Azure/azure-sdk-for-net#2454]
We'd like to optionally block this functionality entirely.  Instead of falling back to local configuration, I'd like to create an overload of GetSetting() that throws if the setting is not found in cloud configuration.  (Currently, all it does is tracewrite the fact that it "fell back" on app.config/web.config configuration.)

However, I cannot find the source to create a pull request. Can somebody please point me towards the source that built this package: https://www.nuget.org/packages/Microsoft.WindowsAzure.ConfigurationManager ?

**Comment by hovsepm** [comment]
@MichaelCommo it is in the master branch. here is the link - https://github.com/Azure/azure-sdk-for-net/tree/master/src/Common/Configuration

**Comment by MichaelCommo** [comment]
@hovsepm: Thanks for the link!  Any tips you can give me W.R.T. pulling/committing changes and getting a new package version published?  This'll be the first time I've contributed to a MS project.

**Comment by MichaelCommo** [comment]
@hovsepm: I have the changes ready to commit; however, I do not have permission to contribute to the master branch.  Is there another branch I should be committing to?  Otherwise, how would I go about contributing?  Thanks in advance!

```
Opening repositories:
V:\DEV\Git\Azure\azure-sdk-for-net
Commit e000a8a4 created locally in repository V:\DEV\Git\Azure\azure-sdk-for-net
Error encountered while pushing to the remote repository: Response status code does not indicate success: 403 (Forbidden).
```

**Comment by MichaelCommo** [comment]
@hovsepm: I'm following the guidelines here: https://azure.github.io/guidelines/
I'll submit a pull request against my fork once I've checked everything off the list. Thanks!

**Comment by MichaelCommo** [comment]
Pull request created. Thank you! 
https://github.com/Azure/azure-sdk-for-net/pull/2459

**Comment by hovsepm** [comment]
@MichaelCommo you should be able to pull the new version. Thanks for contribution!

**Comment by MichaelCommo** [comment]
@hovsepm: I'm getting the error below when trying to use the new features.  Looks like an issue retrieving the error message text from the embedded resource Resources.resx.  I'm looking into it now.  

`[MissingManifestResourceException: Could not find any resources appropriate for the specified culture or the neutral culture.  Make sure "Microsoft.WindowsAzure.Resources.resources" was correctly embedded or linked into assembly "Microsoft.WindowsAzure.Configuration" at compile time, or that all the satellite assemblies required are loadable and fully signed.]
   System.Resources.ManifestBasedResourceGroveler.HandleResourceStreamMissing(String fileName) +441
   System.Resources.ManifestBasedResourceGroveler.GrovelForResourceSet(CultureInfo culture, Dictionary`2 localResourceSets, Boolean tryParents, Boolean createIfNotExists, StackCrawlMark& stackMark) +1201
   System.Resources.ResourceManager.InternalGetResourceSet(CultureInfo requestedCulture, Boolean createIfNotExists, Boolean tryParents, StackCrawlMark& stackMark) +1020
   System.Resources.ResourceManager.InternalGetResourceSet(CultureInfo culture, Boolean createIfNotExists, Boolean tryParents) +48
   System.Resources.ResourceManager.GetString(String name, CultureInfo culture) +558
   Microsoft.Azure.AzureApplicationSettings.GetSetting(String name, Boolean outputResultsToTrace, Boolean throwIfNotFoundInRuntime) +292`

**Comment by MichaelCommo** [comment]
@hovsepm: Was a namespacing issue in the resources designer file.  Bug predates my last pull-request.  I have a new pull request that resolves this issue, as well as new unit tests added to ensure this doesn't happen again.

Pull Request: https://github.com/Azure/azure-sdk-for-net/pull/2495

**Comment by MichaelCommo** [comment]
Feature is available as part of ConfigurationManager v3.2.3
